src/hotspot/share/opto/library_call.cpp

```diff
@@ -5387,6 +5387,10 @@ bool LibraryCallKit::inline_array_partition() {
     const TypeInstPtr* elem_klass = gvn().type(elementType)->isa_instptr();
     ciType* elem_type = elem_klass->const_oop()->as_instance()->java_mirror_type();
     BasicType bt = elem_type->basic_type();
+    // Disable the intrinsic for 64-bit types with AVX2
+    if ((bt == T_LONG || bt == T_DOUBLE) && UseAVX == 2) {
```

Comment:
UseAVX is platform specific and cannot be used in library_call.cpp.

Reply:
Thanks Sandhya, will fix this issue.

Reply:
Thanks Sandhya for suggesting the change to use supports_simd_sort(BasicType bt). Please see the updated code upstreamed.

Reply:
@vamsi-parasa Thanks, your changes look good to me.